ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Settings.hs

```diff
@@ -0,0 +1,336 @@
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QComboBox as QComboBox
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.Util
+
+data Settings =
+  Settings
+    { settings :: QDialog.QDialog
+    , generalButton :: QPushButton.QPushButton
+    , aboutButton :: QPushButton.QPushButton
+    , supportButton :: QPushButton.QPushButton
+    , generalWidget :: QWidget.QWidget
+    , generalSettings :: GeneralSettings
+    , aboutWidget :: QWidget.QWidget
+    , aboutSettings :: AboutSettings
+    , supportWidget :: QWidget.QWidget
+    , supportSettings :: SupportSettings
+    }
+
+data SettingsMode = GeneralMode | AboutMode | SupportMode deriving (Show, Eq, Enum, Bounded)
+
+data GeneralSettings = 
+  GeneralSettings 
+    {  languageSelector :: QComboBox.QComboBox
+    , countervalueSelector :: QComboBox.QComboBox
+    , rateProviderSelector :: QComboBox.QComboBox
+    , themeSelector :: QComboBox.QComboBox
+    }
+data AboutSettings =
+  AboutSettings
+    { releaseNotesButton :: QPushButton.QPushButton
+    , licenseButton :: QPushButton.QPushButton
+    }
+
+data SupportSettings =
+  SupportSettings
+    { faqButton :: QPushButton.QPushButton
+    , clearCacheButton :: QPushButton.QPushButton
+    , reportButton :: QPushButton.QPushButton
+    , downloadLogsButton :: QPushButton.QPushButton
+    , enableRepots :: QCheckBox.QCheckBox
+    , hasAnalytics :: QCheckBox.QCheckBox
+    , resetButton :: QPushButton.QPushButton
+    }
+
+makeLensesWith postfixLFields ''Settings
+
+initSettings :: IO (QDialog.QDialog, Settings)
+initSettings = do
+  settings <- QDialog.new
+  QWidget.resizeRaw settings 776 363
+
+  QObject.setObjectName settings ("settingsDialog" :: String)
+
+  settingsLayout <- QVBoxLayout.new
+  QLayout.setContentsMarginsRaw settingsLayout 0 0 0 0
+  QLayout.setSpacing settingsLayout 0
+
+  labelLayout <- QHBoxLayout.new
+  QLayout.setContentsMarginsRaw labelLayout 0 24 0 24
+  label <- QLabel.newWithText ("SETTINGS" :: String)
+  void $ setProperty label ("styleRole" :: Text) ("dialogHeader" :: Text)
+  QBoxLayout.addWidget labelLayout label
+  void $ QLayout.setWidgetAlignment labelLayout label $ alignHCenter .|. alignVCenter
+
+  QBoxLayout.addLayout settingsLayout labelLayout
+
+  buttonsLayout <-QHBoxLayout.new
+  QLayout.setContentsMarginsRaw buttonsLayout 202 0 202 0
+  void $ QLayout.setWidgetAlignment buttonsLayout label $ alignHCenter .|. alignVCenter
+  generalButton <- QPushButton.newWithText ("GENERAL" :: String)
+  aboutButton <- QPushButton.newWithText ("ABOUT" :: String)
+  supportButton <- QPushButton.newWithText ("SUPPORT" :: String)
+  hSpacer <- QWidget.new
+  for_ [generalButton, aboutButton, supportButton] $ \b ->
+    setProperty b ("styleRole" :: Text) ("settingsTopbarButton" :: Text)
```

Comment:
I believe `setProperty`, `setCheckable` and `addWidget` can be combined into one loop with `do`-block :)